client/hosting/sites/components/sites-dataviews/actions.tsx

```diff
@@ -0,0 +1,233 @@
+import { useI18n } from '@wordpress/react-i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { useMemo } from 'react';
+import SitePreviewLink from 'calypso/components/site-preview-link';
+import {
+	getPluginsUrl,
+	getSettingsUrl,
+	getSiteMonitoringUrl,
+	isCustomDomain,
+	isNotAtomicJetpack,
+	isP2Site,
+	isSimpleSite,
+} from 'calypso/sites-dashboard/utils';
+import { useDispatch as useReduxDispatch } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { launchSiteOrRedirectToLaunchSignupFlow } from 'calypso/state/sites/launch/actions';
+import type { SiteExcerptData } from '@automattic/sites';
+import type { Action } from '@wordpress/dataviews';
+
+export function useActions(): Action< SiteExcerptData >[] {
+	const { __ } = useI18n();
+	const dispatch = useReduxDispatch();
+
+	return useMemo(
+		() => [
+			{
+				id: 'launch-site',
+				label: __( 'Launch site' ),
+				callback: ( sites ) => {
+					dispatch( launchSiteOrRedirectToLaunchSignupFlow( sites[ 0 ].ID, 'sites-dashboard' ) );
+					dispatch( recordTracksEvent( 'calypso_sites_dashboard_site_action_launch_click' ) );
+				},
+				isEligible: ( site ) => {
+					const isLaunched = site.launch_status !== 'unlaunched';
+					const isA4ADevSite = site.is_a4a_dev_site;
+					const isWpcomStagingSite = false; /*useSelector( ( state: AppState ) =>
+						isSiteWpcomStaging( state, site.ID )
+					);*/
+
+					return ! isWpcomStagingSite && ! isLaunched && ! isA4ADevSite;
+				},
+			},
+
+			{
+				id: 'prepare-for-launch',
+				label: __( 'Prepare for launch' ),
+				callback: ( sites ) => {
+					window.location.href = `https://wordpress.com/settings/general/${ sites[ 0 ].ID }`;
+					dispatch(
+						recordTracksEvent( 'calypso_sites_dashboard_site_action_prepare_for_launch_click' )
+					);
+				},
+				isEligible: ( site ) => {
+					const isLaunched = site.launch_status !== 'unlaunched';
+					const isA4ADevSite = site.is_a4a_dev_site;
+					const isWpcomStagingSite = false; /*useSelector( ( state: AppState ) =>
+						isSiteWpcomStaging( state, site.ID )
+					);*/
+
+					return ! isWpcomStagingSite && ! isLaunched && !! isA4ADevSite;
+				},
+			},
+
+			{
+				id: 'settings',
+				label: __( 'Settings' ),
+				callback: ( sites ) => {
+					window.location.href = getSettingsUrl( sites[ 0 ].slug );
+					dispatch( recordTracksEvent( 'calypso_sites_dashboard_site_action_settings_click' ) );
+				},
+			},
+
+			{
+				id: 'hosting',
+				label: __( 'Hosting' ),
+				callback: ( sites ) => {
+					const site = sites[ 0 ];
+					const hasHosting =
+						/* useSafeSiteHasFeature( site.ID, FEATURE_SFTP ) && */ ! site?.plan?.expired;
+
+					window.location.href = hasHosting
+						? `hosting-config/${ site.slug }`
+						: `/hosting-features/${ site.slug }`;
```

Comment:
This should also be `page()` navigation. And the `/hosting-config` path should start with `/`, otherwise it's relative to current URL (`/sites`?) and will probably fail.

Reply:
I'll address these redirects when I'm done with the selectors usage in isEligible. :)